Jun23

fuhlig1 opened the pull request:
Updates several packages to new versions.
Fixing issues probably only relevant on macosx.

fuhlig1 committed:
[legacy] Fix zeromq installation on macosx

The problem with at least macosx 13.5.1 and the corresponding command line
tools is that the gnutls header files are not found since they are not in a
system path.
Add the found include directory explicitly.

fuhlig1 committed:
[legacy] boost: Bump 1.83.0

fuhlig1 committed:
[legacy] fmt: Bump 10.1.1

fuhlig1 committed:
[legacy] fairmq: Bump 1.7.0

fuhlig1 committed:
[legacy] pythia8: Bump 8310

fuhlig1 committed:
[legacy] geant4_vmc: Bump 6-3-p2

fuhlig1 committed:
[legacy] Fix brew command

The previous command to extract the proper macosx sdk doesn't work any longer.
I am not sure if the problem is due to a change in brew, the command line tools
or the macosx version itself. This will be further tested on my other macosx
systems.

dennisklein commented on cmake/legacy.cmake:
```diff
@@ -158,7 +158,7 @@ if(ICU_ROOT)
 endif()
 list(APPEND packages dds)
-set(dds_version "3.6")
+set(dds_version "3.7.22")
```
When we had issues in the past, @AnarManafov adviced against distributing development versions of DDS. Has this changed? Why do you want this new DDS?

fuhlig1 replied:
I simply updated to the latest available version, but i am also fine with 3.6 or 3.7 if uneven numbers doesn't mean development versions.

dennisklein replied:
My latest understanding is that 3.7 (uneven minor version) is development.

fuhlig1 replied:
Went back to version 3.6

AnarManafov replied:
@dennisklein , @fuhlig1, yes this is a developer version. But last couple of years, working with ALICE, I had to release dev versions pretty frequently. According to their policy a patch (dev) version has a shorter release path in comparison to a new major release. I therefor was using this trick and have been releasing only dev versions.  

You can add this one. It is stable.

fuhlig1 replied:
@dennisklein,

should we bump to the latest version if it is stable?

dennisklein replied:
> should we bump to the latest version if it is stable?

yes, fine with me then :)

fuhlig1 replied:
@AnarManafov,

would you consider 3.7.10 also as stable? 
This is the last one not requiring CMake 3.23. We currently only require CMake 3.18.

AnarManafov replied:
@fuhlig1 , yeah, anything tagged was used on  ALICE prod.

fuhlig1 replied:
Bumped DDS to 3.7.10

dennisklein commented on cmake/legacy.cmake:
```diff
@@ -123,31 +123,31 @@ set(boost_features
-  URL_HASH SHA256=1e19565d82e43bc59209a168f5ac899d3ba471d55c7610c677d4ccf2c9c500c0
+  URL_HASH SHA256=6478edfe2f3305127cffe8caf73ea0176c53769f4bf1585be237eb30798c3b8e
   BUILD_IN_SOURCE ON
   PATCH_COMMAND ${patch} -p1 -i "${CMAKE_SOURCE_DIR}/legacy/boost/0004-filesystem-posix-fix-no-at-apis-missing-include.patch"
```
Have you checked that this patch is still needed?

fuhlig1 replied:
I did not check carefully but I can't remember that I have seen a reject in the log file. I will crosscheck again.

The question is if we want to go to the latests boost since this could add problems with the findboost module of cmake!!

fuhlig1 replied:
I removed the patch since the code is already in. I overlooked the output before.

dennisklein reviewed (commented):
Thx! See my inline questions.

fuhlig1 committed:
[legacy] root: Bump 6.28.06

fuhlig1 committed:
[legacy] cmake: Bump 3.27.4 in bootstrap script

fuhlig1 commented:
I had to move for gent4 also to the internal expat on Linux. The compilation of geant4 works without problems but a config file is incorrect when using the system expat. Unfortunately this config file is needed by the vgm and geant4_vmc configuration. 

With this change and the latests CMake 3.27.4 I was able to compile FairSoft jun23 also with Debian10.

dennisklein commented on cmake/legacy.cmake:
```diff
@@ -298,24 +298,22 @@ else()
     "-DGEANT4_USE_GDML=ON"
     "-DGEANT4_USE_OPENGL_X11=OFF"
     "-DGEANT4_USE_RAYTRACER_X11=OFF"
-    "-DGEANT4_USE_PYTHON=ON"
```
Is this intentional? If yes, let's give a reason in the commit msg please

fuhlig1 replied:
Yes this is intentional. The Geant4 python binding were removed from Geant4 and put into an external package. Since the python bindings were only requested and used by the Ship experiment at CERN I removed the python dependency.
The Ship experiment uses a fork of FairSoft since ages.  
I will update the commit message.

fuhlig1 replied:
Done.

fuhlig1 committed:
[legacy] flatbuffers: Do not build tests

There is a known cmake error for some operating systems.
https://github.com/google/flatbuffers/issues/7994

fuhlig1 committed:
[legacy] CMake: Bump minimal required version to 3.18

Adopt latest CMake policies

fuhlig1 committed:
[legacy] dds: Bump 3.7.10

fuhlig1 committed:
[legacy] geant4: Bump 11.1.2

    Use internal expat. When using the system one a config file is wrong which is
    used to build vgm and geant4_vmc. The config for both packages creashes when
    using the system expat.
    The python bindings were moved from Geant4 into an external package so the
    python support ins't needed any longer.

fuhlig1 committed:
[legacy] Update package version info and tested systems

fuhlig1 commented:
I have now updated the information about used package version and started a new list of systems I have tested till now.
